lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -255,6 +257,12 @@ func (handler *ContainersHandlersImpl) RemoveContainerHandler(params containers.
 		case exec.RemovePowerError:
 			return containers.NewContainerRemoveConflict().WithPayload(&models.Error{Message: err.Error()})
 		default:
+			if f, ok := err.(types.HasFault); ok {
+				switch f.Fault().(type) {
+				case *types.HostNotConnected:
+					return containers.NewContainerRemoveDefault(501).WithPayload(&models.Error{Message: "Couldn't remove container.  The ESX host is temporarily disconnected.  Please try again later."})
```

Comment:
Why are we returning a 501 here?

Reply:
That's a good point.  I was going to comment on this but forgot to do so.  Thanks for reminding me.  I think this needs to be code 500.  The swagger definition should also change from

```
					"500": {
						"description": "server error"
					},
```

to

```
					"500": {
						"description": "server error"
						"schema": {
							"$ref": "#/definitions/Error"
						}
					},
```

container.go will also need to change to accomodate the error.

Reply:
We should also use a constant from the `http` pkg here.